src/Rector/MethodCall/LumenRoutesStringMiddlewareToArrayRector.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Laravel\Rector\MethodCall;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\Array_;
+use PhpParser\Node\Expr\ArrayItem;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Scalar\String_;
+use Rector\Core\NodeManipulator\ArrayManipulator;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Laravel\NodeAnalyzer\LumenRouteRegisteringMethodAnalyzer;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\Laravel\Tests\Rector\MethodCall\LumenRoutesStringMiddlewareToArrayRector\LumenRoutesStringMiddlewareToArrayRectorTest
+ */
+final class LumenRoutesStringMiddlewareToArrayRector extends AbstractRector
+{
+    public function __construct(
+        private readonly ArrayManipulator $arrayManipulator,
+        private readonly LumenRouteRegisteringMethodAnalyzer $lumenRouteRegisteringMethodAnalyzer
+    ) {
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Changes middlewares from rule definitions from string to array notation.',
+            [new CodeSample(<<<'CODE_SAMPLE'
+$router->get('/user', ['middleware => 'test']);
+$router->post('/user', ['middleware => 'test|authentication']);
+CODE_SAMPLE
+            , <<<'CODE_SAMPLE'
+$router->get('/user', ['middleware => ['test']]);
+$router->post('/user', ['middleware => ['test', 'authentication']]);
+CODE_SAMPLE)]
+        );
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [MethodCall::class];
+    }
+
+    public function refactor(Node $node): ?Node
+    {
+        if (! $node instanceof MethodCall) {
+            return null;
+        }
+
+        if (! $this->lumenRouteRegisteringMethodAnalyzer->isLumenRoutingClass($node)) {
+            return null;
+        }
+
+        $array = $this->getRouterMethodAttributes($node);
+        if (! $array instanceof Array_) {
+            return null;
+        }
+
+        $arrayItem = $this->findItemInArrayByKey($array, 'middleware');
+        if (! $arrayItem instanceof ArrayItem) {
+            return null;
+        }
+
+        /** @var Node\Scalar\String_|Node\Expr\Array_ $middlewareValue */
+        $middlewareValue = $arrayItem->value;
+        if ($middlewareValue instanceof Array_) {
+            return null;
+        }
+
```

Comment:
Better to add non String_ node value as well to avoid explode non string, eg: variable.

Reply:
Done